### DISCUSSION TITLE
Better explanation  for kubernetes-end-to-end-aws in cdks-examples

### DISCUSSION THREAD

**Discussion by Analect** [cdk8s-team/cdk8s#1327] [Q&A]
Thanks team for your efforts around these examples. I was working through the [kubernetes-end-to-end-aws](https://github.com/cdk8s-team/cdk8s-examples/tree/main/typescript/kubernetes-end-to-end-aws) one today.

It would be great to perhaps have a more detailed blog-post around what's happening behind-the-scenes here. It seems there are >80 resources created as part of the main CloudFormation stack (below) and a few nested ones.

![image](https://github.com/cdk8s-team/cdk8s/assets/4063815/f7450b38-a0d8-4ace-88ee-4e2089ffc68d)

I realise there are some in-line explanations in the [file](https://raw.githubusercontent.com/cdk8s-team/cdk8s-examples/main/typescript/kubernetes-end-to-end-aws/index.ts), but even with that, it feels a bit like a black-box.  It took almost 25 mins to deploy and equally long to destroy.

```
Stack ARN:
arn:aws:cloudformation:eu-west-1:XXXXX:stack/kubernetes-end-to-end/aa1de860-f3cf-11ed-8f60-061c22f4816d

✨  Total time: 1406.82s
```

Certain Lambda functions and even Step Functions were created, and I've no idea why.

![image](https://github.com/cdk8s-team/cdk8s/assets/4063815/32f3c3d9-453b-4873-a43c-0b2c49820918)

![image](https://github.com/cdk8s-team/cdk8s/assets/4063815/31404c94-07dc-4dac-bb23-99c9c2d502e5)

One last thing - the slack invite on https://cdk.dev/ is no longer active.

**Comment by iliapolo** [comment]
@Analect Yeah there's definitely a lot going on here...

All of these resources are coming from the AWS CDK. The VPC construct over there for example creates a lot of resources for you, and the EKS construct is also fairly involved. 

Here are some links that help clarify things:

- https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks-readme.html#architectural-overview
- https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.Vpc.html

We'll see if we can compress those somehow and document them in the example. Any suggestions are welcome :)

Thanks!

**Comment by iliapolo** [comment]
@Analect the slack invite should be fixed now. Thanks for letting us know.
